src/clj/rems/api/services/util.clj

```diff
@@ -1,65 +1,72 @@
 
 (deftest test-forbidden-organization?
-  (testing "for owner, all organizations are permitted"
-    (binding [context/*user* {:organizations [{:organization/id "own organization"}]}
-              context/*roles* #{:owner}]
-      (is (not (forbidden-organization? {:organization/id "own organization"})))
-      (is (not (forbidden-organization? {:organization/id "not own organization"})))
-      (is (not (forbidden-organization? {:organization/id ""})))
-      (is (not (forbidden-organization? {:organization/id nil})))))
+  (let [org-empty {:organization/id ""}
+        org-nobody {:organization/id "organization with no owners" :organization/owners []}
+        org-bob {:organization/id "organization owned by bob" :organization/owners [{:userid "bob"}]}
+        org-carl {:organization/id "organization owned by bob" :organization/owners [{:userid "carl"}]}
+        org-bob-carl {:organization/id "organization owned by bob and carl" :organization/owners [{:userid "bob"} {:userid "carl"}]}]
+    (testing "for owner, all organizations are permitted"
+      (binding [context/*user* {:eppn "owner"}
+                context/*roles* #{:owner}]
+        (is (not (forbidden-organization? org-empty)))
+        (is (not (forbidden-organization? org-nobody)))
+        (is (not (forbidden-organization? org-bob)))
+        (is (not (forbidden-organization? org-carl)))
+        (is (not (forbidden-organization? org-bob-carl)))))
 
-  (testing "for handler, all organizations are permitted"
-    (binding [context/*user* {:organizations [{:organization/id "own organization"}]}
-              context/*roles* #{:handler}]
-      (is (not (forbidden-organization? {:organization/id "own organization"})))
-      (is (not (forbidden-organization? {:organization/id "not own organization"})))
-      (is (not (forbidden-organization? {:organization/id ""})))
-      (is (not (forbidden-organization? {:organization/id nil})))))
+    (testing "for handler, all organizations are permitted"
+      (binding [context/*user* {:eppn "bob"}
+                context/*roles* #{:handler}]
+        (is (not (forbidden-organization? org-empty)))
+        (is (not (forbidden-organization? org-nobody)))
+        (is (not (forbidden-organization? org-bob)))
+        (is (not (forbidden-organization? org-carl)))
+        (is (not (forbidden-organization? org-bob-carl)))))
 
-  (testing "for owner who is also an organization owner, all organizations are permitted"
-    (binding [context/*user* {:organizations [{:organization/id "own organization"}]}
-              context/*roles* #{:owner :organization-owner}]
-      (is (not (forbidden-organization? {:organization/id "own organization"})))
-      (is (not (forbidden-organization? {:organization/id "not own organization"})))
-      (is (not (forbidden-organization? {:organization/id ""})))
-      (is (not (forbidden-organization? {:organization/id nil})))))
+    (testing "for owner, all organizations are permitted"
```

Comment:
This feels like duplicate of the previous. Should something change in the description or the content?

Reply:
woops, done